### PR TITLE
add max depth opt

### PR DESCRIPTION
**Pull request by roger-liu-bina**
Add max depth for a scan in order to avoid mem blowup. Also, give warnings to reads without cigar.

### CODE REVIEW

**Comment by msahraeian** [comment]
@roger-liu-bina thanks. It sounds good to me. Can you make the default as 40000?
@marghoob please review.

**Review by msahraeian — CHANGES_REQUESTED** [review]
Sounds good. Please update default to 40000

**Comment by roger-liu-bina** [comment]
Sure. Now it is 40000.

File: neusomatic/include/Options.h [hunk 1]
```diff
@@ -45,7 +46,8 @@ namespace neusomatic {
     std::cerr<< "-o/--out_count_file,                  output count file path,                                                required.\n";
     std::cerr<< "-w/--window_size,                     window size to scan the variants,                                      default is 15\n";
     std::cerr<< "-y/--fully_contained,                 if this option is on. A read has to be fully contained in the region,  default is False\n";
-    std::cerr<< "-t/--num_threads,                      numbef or thread used for building the count matrix,                   default is 4\n";
+    std::cerr<< "-t/--num_threads,                     numbef or thread used for building the count matrix,                   default is 4\n";
+    std::cerr<< "-d/--max_depth,                       maximum depth for building the count matrix,                           default is 20,000\n";
```
Comment (msahraeian) [inline neusomatic/include/Options.h]: @roger-liu-bina please fix this too to 40,000

**Review by msahraeian — APPROVED** [review]

File: neusomatic/include/Options.h [hunk 2]
```diff
@@ -22,11 +22,12 @@ namespace neusomatic {
...
     {"window_size",                     required_argument,      0,        'w'},
     {"num_threads",                     required_argument,      0,        't'},
+    {"max_depth",                       required_argument,      0,        'd'},
     {0, 0, 0, 0} // terminator
   };
 
   //remove calculate_qual_stat
-  const char *short_options = "v:b:L:r:q:f:o:w:a:t:cy";
+  const char *short_options = "v:b:L:r:q:f:o:w:a:t:d:cy";
   //const char *short_options = "v:b:L:r:q:f:o:w:a:t:y";
```
Comment (marghoob) [inline neusomatic/include/Options.h]: Please remove this commented out line.

File: neusomatic/include/Options.h [hunk 3]
```diff
@@ -45,7 +46,8 @@ namespace neusomatic {
     std::cerr<< "-o/--out_count_file,                  output count file path,                                                required.\n";
     std::cerr<< "-w/--window_size,                     window size to scan the variants,                                      default is 15\n";
     std::cerr<< "-y/--fully_contained,                 if this option is on. A read has to be fully contained in the region,  default is False\n";
-    std::cerr<< "-t/--num_threads,                      numbef or thread used for building the count matrix,                   default is 4\n";
+    std::cerr<< "-t/--num_threads,                     numbef or thread used for building the count matrix,                   default is 4\n";
```
Comment (marghoob) [inline neusomatic/include/Options.h]: "numbef or" --> number of

File: neusomatic/include/Options.h [hunk 4]
```diff
@@ -45,7 +46,8 @@ namespace neusomatic {
...
+    std::cerr<< "-d/--max_depth,                       maximum depth for building the count matrix,                           default is 40,000\n";
```
Comment (marghoob) [inline neusomatic/include/Options.h:48]: Do we parse commas in the option? If not, better to change "40,000" to "40000" to avoid confusion.

**Review by marghoob — COMMENTED** [review]
Other than the minor comments, the pull request looks good to me.

File: neusomatic/include/Options.h (hunk 4 above)
Comment (roger-liu-bina) [reply neusomatic/include/Options.h:48]: That is only for display. This string is not parsed.

**Comment by roger-liu-bina** [comment]
@msahraeian read for merge

**Pull request merged**